src/features/registration/registration-form.tsx

```diff
@@ -0,0 +1,326 @@
+    }, [allUploadProgress])
+
+    const handleSubmit = async (e: React.FormEvent) => {
+        setIsSubmitting(true);
+        setDisabled(true);
+        e.preventDefault();
+        await invoke<string>('get_hostname')
+            .then(async hostname => await postRegistration(hostname))
+            .catch(error => {
+                console.error(error)
+            })
+            .finally(() => setIsSubmitting(false));
+    };
+
+    const postRegistration = async (machineName: string) => {
+        if (!state.current?.path) return;
+        const pushedDir = state.current?.path;
+        const auth = await settings.getAuthResponse();
+        if (!auth || loggedOut) return Promise.reject('Not logged in');
+
+        const id = uuidv7().toString();
+
+        const transfer = uploadToS3(id).catch(error => {
+            handleError('Fikk ikke lastet opp filene', undefined, error);
+            return Promise.reject(error);
+        });
+
+        const numberOfPagesTransferred = await transfer;
+
+        const accessToken = await invoke('get_papi_access_token').catch(error => {
+            handleError('Kunne ikke hente tilgangsnøkkel for å lagre objektet i databasen.', undefined, error);
+            return Promise.reject(error);
+        });
+
+        const materialTypeDTO = Object.keys(MaterialType).find((key: string) => MaterialType[key as keyof typeof MaterialType] === materialType)
+
+        return await fetch(`${papiPath}/v2/item`, {
+            method: 'POST',
+            headers: {
+                'Authorization': 'Bearer ' + accessToken,
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(new TextInputDto(
+                id,
+                materialTypeDTO!,
+                auth.userInfo.name,
+                fraktur ? 'FRAKTUR' : 'ANTIQUA',
+                sami ? 'SME' : 'NOB',
+                machineName,
+                name,
+                numberOfPagesTransferred
+            ))
+        })
+            .then(async response => {
+                if (response.ok) {
+                    void deleteDir(pushedDir);
+                    removeErrorMessage();
+                    displaySuccessMessage(await response.json() as TextItemResponse);
+                } else {
+                    console.error(response);
+                    handleError(undefined, response.status);
+                }
+            })
+            .catch(error => {
+                handleError();
+                return Promise.reject(error);
+            });
+    };
+
+    const uploadToS3 = async (id: string): Promise<number> => {
+        const filesPath = await settings.getScannerPath();
+        if (filesPath === state.current?.path) {
+            return Promise.reject('Cannot move files from scanner dir');
+        }
+
+        const materialTypeEnum = Object.keys(MaterialType).find(key => MaterialType[key as keyof typeof MaterialType] === materialType);
```

Comment:
Siden denne er litt fancy-schmancy og brukes to steder i denne komponenten (her og linje 114) så kunne det kanskje vært dratt ut i en liten hjelpemetode.